src/RobotSimulator.elm

```diff
@@ -0,0 +1,305 @@
+module Main exposing (main, update, view)
+
+import Browser
+import Browser.Events
+import Html exposing (..)
+import Html.Attributes exposing (style)
+import Html.Events exposing (onClick)
+import Json.Decode as Decode
+import Time
+
+
+screenWidth =
+    500
+
+
+screenHeight =
+    500
+
+
+shotVelocity =
+    Vector 1 1
+
+
+main =
+    Browser.element
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Vector =
```

Comment:
I'd parameterize (`Vector number`) - there's no reason to be limited to `Float`, any `number` can work.